### PR TITLE
[Easy][BE]: Enable clang-tidy check for duplicate includes

### PR DESCRIPTION
**Pull request by Skylion007**
Adds a clang-tidy check to flag duplicate include files

### CODE REVIEW

File: torch/csrc/Size.cpp [hunk 1]
```diff
@@ -167,8 +166,8 @@ static PyObject* wrap_tuple_fn(Args... args) {
 // We use an anonymous namespace instead of static to work around
 // (what @peterjc123 think is) a bug in Visual Studio
 namespace {
-auto sq_concat = PyTuple_Type.tp_as_sequence->sq_concat;
-auto sq_repeat = PyTuple_Type.tp_as_sequence->sq_repeat;
+auto sq_concat = PyTuple_Type.tp_as_sequence -> sq_concat;
```
Comment (albanD) [inline torch/csrc/Size.cpp:169]: This is a bit weird no?

**Review by albanD — APPROVED** [review]
Thanks!

File: torch/csrc/Size.cpp (hunk 1 above)
Comment (Skylion007) [reply torch/csrc/Size.cpp]: clang-format did it. *shrugs

**Review by malfet — APPROVED** [review]

**Comment by malfet** [comment]
@jeanschmidt I think signal you are looking at is wrong, it's impossible to break anything with this change as it is a no-op

**Comment by jeanschmidt** [comment]
@malfet could be, but then I reproduced the issue internally multiple times. Do you believe we should re-merge the PR?

**Comment by jeanschmidt** [comment]
After discussing with @malfet, we agreed that the best path forward is to merge this back, so we can work internally towards fixing what needs to be fixed without blocking OSS development.

As with this revert and reland it should be sufficient to unblock internal diff train to continue imports due the internal code freeze.

**Pull request closed without merging**